AK/BinarySearch.h

```diff
@@ -38,17 +38,23 @@ int integral_compare(const T& a, const T& b)
 }
 
 template<typename T, typename Compare>
-T* binary_search(T* haystack, size_t haystack_size, const T& needle, Compare compare = integral_compare, int* nearby_index = nullptr)
+T* binary_search(T* haystack, size_t haystack_size, const T& needle, Compare compare = integral_compare, size_t* nearby_index = nullptr)
 {
-    int low = 0;
-    int high = haystack_size - 1;
+    if(!haystack || !haystack_size)
```

Comment:
Please use `clang-format`.
Also please add a test case to TestBinarySearch.cpp that fails without this fix :)